### ISSUE TITLE
Accessing Contact Entity Image

### ISSUE THREAD

**Issue by topness-msft** [MicrosoftDocs/powerapps-docs#154]
I'm trying to duplicate the contact embedded canvas app as shown in this article, with contact images.  Is the solution file for that available in Github?  I'm unable to get the Contact EntityImage to render properly.  In Canvas apps, I noticed that EntityImage_URL was relative and prepended the app's ...dynamics.com URL.  With this, contact images render in the canvas app designer.  However images don't show when the canvas app control is embedded in a model drive app form.  I validated that the browser's cross-origin blocking was not preventing it from loading the image.  Any guidance?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 471d1aeb-ee46-d077-1819-09d37014e41d
* Version Independent ID: bb500f3e-4b1f-6077-c24d-bd18d0e88035
* Content: [Embed a canvas app on a model-driven form - PowerApps](https://docs.microsoft.com/en-us/powerapps/maker/model-driven-apps/embed-canvas-app-in-form)
* Content Source: [powerapps-docs/maker/model-driven-apps/embed-canvas-app-in-form.md](https://github.com/MicrosoftDocs/powerapps-docs/blob/live/powerapps-docs/maker/model-driven-apps/embed-canvas-app-in-form.md)
* Service: **crm-online**
* GitHub Login: @aneesmsft
* Microsoft Alias: **matp**

**Comment by mostness** [comment]
Hi All - if there is code for the canvas control that's used in this documentation and it can be shared, I'd be happy to explore it and document my findings.

**Comment by aneesmsft** [comment]
For my app I am not loading the images from CDS directly. I put the images in SharePoint to show integration with other cloud services. For you this could be any other cloud service that you can connect to using Canvas apps. 

In my app I find (lookup) the SharePoint list item and load the corresponding image. The expression looks something like this: 
LookUp('Contacts List', 'Primary Email' = ThisItem.EMailAddress1).'Profile Picture'
'Contacts List' is the SharePoint list
'Primary Email' and 'Profile Picture' are SharePoint list items.
'EmailAddress1' is a field on the Contact entity.
Basically the expression says, match the email to get the record and then get the picture from that.

An alternative could be to put images in Azure blob storage, save the corresponding URLs in CDS and load the images that way. See this blog post for more details: [Upload files from PowerApps using the Azure Blob Storage connector](https://powerapps.microsoft.com/blog/upload-files-from-powerapps-using-the-azure-blob-storage-connector/)

**Comment by mostness** [comment]
Thanks so much for the detailed response.  I understand how you approached it now, it makes sense.  Good suggestions.

**Comment by foosfam** [comment]
Is the code available for this???  How do you put together a demo like this and not share the code so we can see what you did?  Come on Microsoft....

**Comment by aneesmsft** [comment]
@foosfam, this is not a sample app and therefore there is no official code or app shared. I shared all the details above for the way I build this example. If you tried that and still were not able to I am happy to help.

**Comment by Mattp123** [comment]
Closing issue.
